portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +26,48 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<Comments> comments = new ArrayList<>();
```

Comment:
Generally, if you don't have a reason to **need** an ArrayList, you gain flexibility by making the variable type List<Comments> instead. The object creation needs to be the concrete type, though. What this does is allow you to change the list type with only the need to change the creation type.